### ISSUE TITLE
Page glitch when loading "Support" link

### ISSUE THREAD

**Issue by dzianiskuzmich** [calcom/cal.com#9342]
### Issue Summary

When the unauthorized User taps "Support" link from the footer there is a glitch on a background before the "Support" page is loaded.

### Steps to Reproduce

1. Go to https://cal.com
2. Scroll down to the footer of the page --> Select "Support"

Expected Result: "Support" page is loaded
Actual Result: A glitch with an Error appears on a background before the "Support" page is loaded (see video attached)

https://github.com/calcom/cal.com/assets/135644174/0c27d741-b721-4afe-9f49-438502557ff7

### Technical details

- Browser Google Chrome version 113.0.5672.126. MacOS Ventura 13.0 (22A380)

**Comment by a0m0rajab** [comment]
I could reproduce this, one thing to consider is to be logged off (or using incognito)

**Comment by a0m0rajab** [comment]
Related code: 

https://github.com/calcom/cal.com/blob/c182091fa9d84c2b4c20263ac7d2691f8c02ff58/apps/web/next.config.js#L323-L325

The support link is redirecting to the app itself, a check for the person if they are logged in or not would solve this issue, 

It might worth to check what does `openIntercom=true` do

**Comment by a0m0rajab** [comment]
Intercom related code: 

https://github.com/calcom/cal.com/blob/c182091fa9d84c2b4c20263ac7d2691f8c02ff58/apps/web/pages/event-types/index.tsx#L792-L796

A quick solution would be checking if the user logged in or not, then doing the redirection.
If they are logged it, it would go to the same event-types page, 
otherwise it would go to auth (https://app.cal.com/auth/login) page.  
A great solution would be implementing this on the home page
Another possible solution would be having a support page with FAQ questions and the intercom. 

To implement the intercom on the auth page, you need to get the intercom module and code.

**Comment by The-Robin-Hood** [comment]
@a0m0rajab you working on this mate ??

**Comment by a0m0rajab** [comment]
Yes, thanks for asking, I just did the initial research and will do my commits and tests in the next few hours. 

One final note, is that the app is using `[next-auth](https://next-auth.js.org/getting-started/client#usesession)` which could be used to check if someone is logged in or not.

**Comment by The-Robin-Hood** [comment]
The behaviour you're seeing is due to the use of an `anonymous_id` in the request URL 
"https://api-iam.intercom.io/messenger/web/open"
When the `anonymous_id` remains the same, intercom.io retrieves the associated name along with the entire messages.

Hope the behaviour is not suppose to be like this. @PeerRich 

https://github.com/calcom/cal.com/assets/32297581/1c31ebad-c8d6-4122-994c-dc6c7d2a005c

**Comment by a0m0rajab** [comment]
There is a support page for the app which is: https://cal.com/support 
but the link redirects to the app,  I am not sure how to go, plus the 404 is good for support if we add the intercom interface there. 

https://nextjs.org/docs/pages/api-reference/next-config-js/redirects#header-cookie-and-query-matching

I think I will go with the support page that looks like 404  but add intercom as well there.

**Comment by PeerRich** [comment]
we need the login so we can correctly identify the person for intercom. we could, however, make the "redirect" sequence a bit nicer and get rid of the "unauthorized" screen and go straight to login, no?

**Comment by PeerRich** [comment]
also, i think intercom should ONLY load when logged in, and not during the auth screen

**Comment by PeerRich** [comment]
> I also tried to recreate this and I noticed that even If i am logged out, My name is still there . It should not be like that. **Do you have any idea why this is happening?**
> 
> ![image](https://user-images.githubusercontent.com/106580067/243709330-79d9c453-387b-4bd7-a3e5-36dbaa565ff0.png)

this is because you logged in before and there is an intercom cookie in your browser

**Comment by a0m0rajab** [comment]
@PeerRich would not asking for the person email in the intercom be enough for you to provide support? I am thinking about it as someone who is from a company and thinking to shift to your product.

**Comment by PeerRich** [comment]
> @PeerRich would not asking for the person email in the intercom be enough for you to provide support? I am thinking about it as someone who is from a company and thinking to shift to your product.

no, because a lot of people enter the wrong email that they are not using for their cal.com account. its really hard to do support without the full context
